spec/serializable_spec.cr

```diff
@@ -172,6 +195,17 @@ describe "DB::Serializable" do
     expect_model("1,a", ModelWithJSON, {c0: 1, c1: "a"})
   end
 
+  it "should initialize a model with an enum property" do
+    expect_model("1,Baz,LOL", ModelWithEnum, {
```

Comment:
It'd be nice to have a similar test but with the int values of the enum. @jgaskins do you want to take it, or do you prefer us to do it? Thanks for your PR!

Reply:
@beta-ziliani Nice catch, I thought I included a test for that. I'll go ahead and add it.

Reply:
Looks like this isn't as cut-and-dry as I thought it would be. The `DummyDriver` returns everything as strings, so the `case` statement always ends up in the `String` code path.

I tried modifying [this line](https://github.com/crystal-lang/crystal-db/blob/07c68d38e49e06ae4a56c4060da4bd2748df6701/spec/dummy_driver.cr#L190) to detect numbers (returns `n.to_i64? || n` instead of `n`), but it broke [this spec](https://github.com/crystal-lang/crystal-db/blob/07c68d38e49e06ae4a56c4060da4bd2748df6701/spec/database_spec.cr#L11). That's the only spec that breaks, but I can change that to a non-numeric value if that makes the most sense.

Any guidance on this one?

Reply:
@beta-ziliani I added ae9a7a82505220399da88840bd9f0a0ab9e6925a to do this. Let me know if updating that other spec is too broad a change and I'll back it out.

Reply:
Ugh, I guess we can put it as `pending` for the moment and leave it for the future to have a better stub for the ResultSet that includes column types. WDYT?

Reply:
@beta-ziliani With https://github.com/crystal-lang/crystal-db/commit/ae9a7a82505220399da88840bd9f0a0ab9e6925a this looks good to me, actually.

Reply:
That makes sense. I'll revert https://github.com/crystal-lang/crystal-db/commit/ae9a7a82505220399da88840bd9f0a0ab9e6925a and add a pending spec for it.

Reply:
Wait. What's the problem with the current solution? I think it works. The number/string selection might not be very elegant, but IMO it's okay. We can't use strings that parse as integers in tests. But that's okay. There's no need to do that because we can chose the test values freely.

Reply:
OK then. The mockup ResultSet is all about parsing strings after all, so I guess you're right. @jgaskins let's leave it as is.

Reply:
Fantastic. I was concerned it might be too broad a change to modify the test harness to accommodate a tiny (though useful) feature, but I'm glad to know it's okay. Thanks friends!

Reply:
I suppose we'll have to see whether that breaks anything in other db drivers that use the same test harness.